sklearn/decomposition/online_lda.py

```diff
@@ -0,0 +1,522 @@
+        """
+
+        X = self._to_csr(X)
+        n_docs, n_vocabs = X.shape
+
+        if not hasattr(self, 'components_'):
+            raise NotFittedError(
+                "no 'components_' attr in model. Please fit model first.")
+        # make sure vocabulary size is the same in fitted model and new doc
+        # matrix
+        if n_vocabs != self.n_vocabs:
+            raise ValueError(
+                "feature dimension(vocabulary size) does not match.")
+
+        gamma, _ = self._e_step(X, False)
+
+        if self.normalize_doc:
+            gamma /= gamma.sum(axis=1)[:, np.newaxis]
+
+        return gamma
+
+    def _approx_bound(self, X, gamma, sub_sampling):
+        """
+        calculate approximate bound for data X and topic distribution gamma
+
+
+        Parameters
+        ----------
+        X: sparse matrix, [n_docs, n_vocabs]
+
+        gamma: array, shape = [n_docs, n_topics]
+            document distribution (can be either normalized & un-normalized)
+
+        sub_sampling: boolean, optional, (default: False)
+            Compensate for the subsampling of the population of documents
+            set subsampling to `True` for online learning
+
+        Returns
+        -------
+        score: float, score of gamma
+        """
+        X = self._to_csr(X)
+        n_docs, n_topics = gamma.shape
+        score = 0
+        Elogtheta = _dirichlet_expectation(gamma)
+
+        X_data = X.data
+        X_indices = X.indices
+        X_indptr = X.indptr
+
+        # E[log p(docs | theta, beta)]
+        for d in xrange(0, n_docs):
+            ids = X_indices[X_indptr[d]:X_indptr[d + 1]]
+            cnts = X_data[X_indptr[d]:X_indptr[d + 1]]
+            id_length = len(ids)
+            phinorm = np.zeros(id_length)
+            for i in xrange(0, id_length):
+                temp = Elogtheta[d, :] + self.Elogbeta[:, ids[i]]
+                tmax = temp.max()
+                phinorm[i] = np.log(np.sum(np.exp(temp - tmax))) + tmax
+            score += np.sum(cnts * phinorm)
+
+        # E[log p(theta | alpha) - log q(theta | gamma)]
+        score += np.sum((self.alpha - gamma) * Elogtheta)
+        score += np.sum(gammaln(gamma) - gammaln(self.alpha))
+        score += np.sum(
+            gammaln(self.alpha * self.n_topics) - gammaln(np.sum(gamma, 1)))
+
+        # Compensate for the subsampling of the population of documents
+        # E[log p(beta | eta) - log q (beta | lambda)]
+        score += np.sum((self.eta - self.components_) * self.Elogbeta)
+        score += np.sum(gammaln(self.components_) - gammaln(self.eta))
+        score += np.sum(gammaln(self.eta * self.n_vocabs)
+                        - gammaln(np.sum(self.components_, 1)))
+
+        # Compensate for the subsampling of the population of documents
+        if sub_sampling:
+            doc_ratio = float(self.n_docs) / n_docs
+            score *= doc_ratio
+
+        return score
+
+    def preplexity(self, X, gamma, sub_sampling=False):
```

Comment:
It would make sense to have a ``score`` method based on ``transform`` and ``perplexity``, right?